### DISCUSSION TITLE
How to apply alias for @vercel/node

### DISCUSSION THREAD

**Discussion by Sh031224** [vercel/vercel#8967] [Help] [answered]
I am trying to build and deploy express server as @vercel/node.

Also, I tried to apply module-alias, but it runs fine locally, but when I connect after building, the following error occurs.

```
2022-11-24T06:35:50.968Z	undefined	ERROR	Error: Unable to find package.json in any of:
[/var/task/node_modules/.pnpm/module-alias@2.2.2,
```

My config is like below.
Until module-alias was applied, it worked without error even when connected after building.

```json
// vercel.json
{
  "version": 2,
  "builds": [
    {
      "src": "src/index.ts",
      "use": "@vercel/node"
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "src/index.ts"
    }
  ]
}
```

```json
// package.json
{
  ...
  "_moduleAliases": {
    "@": "src"
  }
}
```

**Comment by amyegan** [comment]
Hi @Sh031224! Can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) to give us a closer look at the problem?

**Reply by Sh031224** [reply]
I tried to apply the `module-alias` to my project, but it wasn't applied and I was getting an error.

I realized through vercel query that my vercel.json configuration is legacy.

So I updated vercel.json with the latest configuration and built it through babel.
And then the problem was solved.

Records for understanding history.

- before
with module-alias
```
// vercel.json
{
  "version": 2,
  "builds": [
    {
      "src": "src/index.ts",
      "use": "@vercel/node"
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "src/index.ts"
    }
  ]
}
```

- after
with babel-plugin-module-resolver
```
// vercel.json
{
  "functions": {
    "api/index.js": {
      "runtime": "@vercel/node@2.7.0"
    }
  },
  "rewrites": [{ "source": "/api/(.*)", "destination": "/api" }]
}
```

**Comment by swarnava** [comment]
Our team has also reached out to you via email. Ticket #00109551!

**Comment by Sh031224** [comment, marked as answer]
I tried to apply the `module-alias` to my project, but it wasn't applied and I was getting an error.

I realized through vercel query that my vercel.json configuration is legacy.

So I updated vercel.json with the latest configuration and built it through babel.
And then the problem was solved.

Records for understanding history.

- before
with module-alias
```
// vercel.json
{
  "version": 2,
  "builds": [
    {
      "src": "src/index.ts",
      "use": "@vercel/node"
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "src/index.ts"
    }
  ]
}
```

- after
with babel-plugin-module-resolver
```
// vercel.json
{
  "functions": {
    "api/index.js": {
      "runtime": "@vercel/node@2.7.0"
    }
  },
  "rewrites": [{ "source": "/api/(.*)", "destination": "/api" }]
}
```
